### PR TITLE
Disable file attribute masking on Windows 10

### PR DESCRIPTION
**Pull request by drwtsn32x**
Fix (partial?) for Issue https://github.com/duplicati/duplicati/issues/3411

Discussion here:  https://forum.duplicati.com/t/exclude-offline-files-broken-with-onedrive/5196/13

Root cause:
Starting with Windows 10 1803, the operating system may mask the process's view of some file attributes such as reparse, offline, and sparse.

See:
https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/ntifs/nf-ntifs-rtlqueryprocessplaceholdercompatibilitymode

Proposed change below disables this masking.

Demonstration:

Screen shot showing that the "Offline.txt" file has been flagged in OneDrive as available online only. You can see the cloud icon for its status:
Capture

Now when I run test-filter the file is properly excluded.  Before it was included due to the file attribute masking:
```
Duplicati.CommandLine.exe test-filter "C:\Users\xxx\OneDrive\Test" --exclude-files-attributes=Offline

Including source path: C:\Users\xxx\OneDrive\Test\
Excluding path due to attribute filter: C:\Users\xxx\OneDrive\Test\Offline.txt
Including path as no filters matched: C:\Users\xxx\OneDrive\Test\Online.txt
Matched 1 files (6 bytes)
```

For a full fix of issue 3411 maybe Duplicati defaults need to change to skip Offline files by default? If decision to go that route, I think it should be in a separate PR.

### CODE REVIEW

File: Duplicati/Library/Main/ProcessController.cs [hunk 1]
```diff
@@ -234,6 +234,35 @@ private void ActivateBackgroundIOPriority()
...
+            // See https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/ntifs/nf-ntifs-rtlqueryprocessplaceholdercompatibilitymode
+
+            if (Platform.IsClientWindows)
+            {
+                try
+                {
+                    var currentPcm = Win32.RtlQueryProcessPlaceholderCompatibilityMode();
+                    if (currentPcm == Win32.PHCM_VALUES.PHCM_DISGUISE_PLACEHOLDER)
+                    {
+                        Win32.RtlSetProcessPlaceholderCompatibilityMode(Win32.PHCM_VALUES.PHCM_EXPOSE_PLACEHOLDERS);
```
Comment (verhoek) [inline Duplicati/Library/Main/ProcessController.cs]: Just call the setter without the if, and remove the getter. Then we also don't need the import of the getter.

Comment (drwtsn32x) [reply Duplicati/Library/Main/ProcessController.cs]: Yep I can do that if you prefer. I will submit an update later today.  Thanks

**Comment by warwickmm** [comment]
@drwtsn32x, do you think this should be included in the next beta, or should we wait until after pull request #3992 is merged?

**Comment by drwtsn32x** [comment]
@verhoek - changed code as you requested!

**Comment by drwtsn32x** [comment]
@warwickmm - 3992 won't make it to the next beta? Bummer, I think that one is valuable and should be included, unless you think it needs more testing.

This PR here is pretty basic and should be very low risk to merge. That being said it probably doesn't help or affect most users.  I don't know how many are trying to exclude Offline files.

I still think we need to discuss if Offline files should be excluded by default, but I'd like that to be out of scope for this PR.

**Comment by warwickmm** [comment]
Sorry, I was suggesting that #3992 be merged first so that we can optionally create a release from there before other changes are integrated.  #3992 should definitely be part of the next release!

**Comment by drwtsn32x** [comment]
Ok, gotcha!  I think both can be merged for the next release. If this one needs Canary testing then it can wait.  I know one user is affected by the bug that fixes this but haven't heard of any other people complaining.

**Comment by verhoek** [comment]
Both could go in, I will review #3992 later, this change here looks rather harmless.

**Pull request merged**